lib/spack/llnl/util/filesystem.py

```diff
@@ -472,3 +474,110 @@ def find_library_path(libname, *paths):
         if os.path.exists(library):
             return path
     return None
+
+# Utilities for libraries
+
+
+class LibraryList(collections.Sequence):
+    """Manages a list of absolute paths to libraries"""
+    def __init__(self, libraries):
+        self.libraries = list(libraries)
+
+    @property
+    def directories(self):
+        return [os.path.dirname(x) for x in self.libraries]
+
+    @property
+    def libnames(self):
+        return [os.path.basename(x).split('.')[0] for x in self.libraries]
+
+    @property
+    def names(self):
+        return [x[3:] for x in self.libnames]
+
+    @property
+    def search_flags(self):
+        return ['-L' + x for x in set(self.directories)]
+
+    @property
+    def link_flags(self):
+        return ['-l' + name for name in self.names]
+
```

Comment:
i would add one more property for convenience:  join what you call `search_flags` and `link_flags` into a single string `-L/path/to/dir1 -L/path/to/dir2 -ld1 ld11 -ld2  -lbar -lbaz`